### ISSUE TITLE
Replace Locust load generator's httpUser with fastHttpUser

### ISSUE THREAD

**Issue by antokarant** [GoogleCloudPlatform/microservices-demo#1973]
I have a remote kubernetes cluster and trying to use locust either from the load generator pod targeting the frontend or locally targeting the port-forwarded frontend causes an error in the two endpoints mentioned above. Everything else works fine. Creating requests to these endpoints either through postman or through curl does not generate the same error and my first guess was that it had something to do with the session-id cookie. I tried to add to the requests a generated cookie but it did not resolve my issue. The cluster runs the kustomize version so that i can also use Istio. I also tried https requests but that caused an issue in all the endpoints.

![image](https://github.com/GoogleCloudPlatform/microservices-demo/assets/64793891/0f70110b-c252-429f-98f2-6b6b4745d602)
![image](https://github.com/GoogleCloudPlatform/microservices-demo/assets/64793891/78dc3ce8-765c-435a-ad2e-f1b52aa70252)

**Comment by minherz** [comment]
@antokarant thank you for reporting this to us. We will review your use case and see if we can assist. Can you please share the full request and response bodies for the aforementioned requests?

**Comment by antokarant** [comment]
Hello @minherz thanks for your reply. The requests are the ones that already exist in the [locustfile.py](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/loadgenerator/locustfile.py).

When trying with postman or curl I just create a default post request to /setCurrency or /cart/checkout and it works fine without any headers or body.

I also tried it with python requests library and no problem occurred with it either.

In the meanwhile, I found that by replacing HttpUser with FastHttpUser the problem seems to go away so it might be something in the configuration of HttpUser that causes the issue.(a default request flag maybe)

**Comment by NimJay** [comment]
@antokarant thanks for the very clear description of your issue!
@minherz thanks for the quick response and looking into this!

@antokarant:
* I'm glad you were able to get it working with `FastHttpUser` instead of the plain [`HttpUser`](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/loadgenerator/locustfile.py#L79).
* According to [this doc](https://github.com/locustio/locust/blob/master/docs/increase-performance.rst):
    *  `FastHttpUser` uses [`geventhttpclient`](https://github.com/geventhttpclient/geventhttpclient)
    *  `HttpUser` uses `python-requests`
* So it's likely (as you suggested) that there is some difference in the HTTP requests (e.g., default set of headers) (used by `python-requests` and [`geventhttpclient`](https://github.com/geventhttpclient/geventhttpclient)).
* If you do figure out that difference (and specifically what's causing the 405), let us know — especially if it's something that needs to be fixed on Online Boutique's (microservices-demo's) end.

**Questions:** 
1. Which version of Online Boutique are you using? The current `main` branch? Or a specific tag?
2. Are you able to see exactly what the failing /setCurrency and /cart/checkout requests look like (i.e., list of headers, the body, referrer)? If you're building the microservice Docker images from scratch, perhaps you could add some logs into [the frontend](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/frontend/main.go#L140) to see exactly what the requests to /setCurrency and /cart/checkout look like?

Thanks!

**Comment by antokarant** [comment]
@NimJay thanks for your reply.

- I am using the main branch and I deployed the project using the Kustomize tool on a local cluster as I also need it to be istio-mesh enabled.

- To answer your second question, I could not exactly see the request that caused the error. I tried different headers using python-requests but I could not find the solution.

**Question**:
I found out that kustomize uses container images already on google cloud so I can not find a way to change the requests inside the load-generator pod(the proposed fix using FastHttpUser was tested while using Locust from my local machine to access the cluster). I was wondering if there is a way to deploy using skaffold but also to have istio installed. Or if there is another workaround to fix my issue.

Thanks!
